src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1449,12 +1449,14 @@ class DocBaseViewer extends BaseViewer {
     }
 
     /**
-     * Determines if the thumbnails sidebar is toggled based on the value from the cache
-     * as well as defaulting to true if there is no cache entry for this file id.
-     * @return {boolean} Whether thumbnails is toggled open or not
+     * Determines if the thumbnails sidebar should be toggled
+     * @return {boolean} Whether thumbnails should be toggled open or not
      */
     shouldThumbnailsBeToggled() {
         const cachedToggledState = this.getCachedThumbnailsToggledState();
+        // `pdfViewer.pagesCount` isn't immediately available after pdfViewer.setDocument()
+        // is called, but the numPages is available on the underlying pdfViewr.pdfDocument
+        const { numPages } = this.pdfViewer && this.pdfViewer.pdfDocument;
```

Comment:
You should be able to grab the num pages directly off of the PDF viewer.

`const { pagesCount } = this.pdfViewer;`